[BUG] Action Fails to Deploy With No Output

**Describe the bug**
The [generated deployment command](https://github.com/sws2apps/firebase-deployment/blob/main/src/index.js#L28) does not succeed.  It runs in a separate process, and the underlying logs are not displayed.

**To Reproduce**
Steps to reproduce the behavior:

1. Here is the GitHub Action Workflow I am using:

```yaml
name: Deploy to Firebase

on:
  push:
    branches:
      - main

permissions: read-all

jobs:
  main:
    name: Build and Deploy
    runs-on: ubuntu-latest

    permissions:
      contents: "read"
      id-token: "write"

    steps:
      - name: Checkout
        uses: actions/checkout@v3

      - name: Deploy to Firebase
        uses: sws2apps/firebase-deployment@main #consider using pin for dependabot auto update
        with:
          workloadIdentityProvider: ${{ secrets.WORKLOAD_IDENTITY_PROVIDER }}
          serviceAccount: ${{ secrets.SERVICE_ACCOUNT_EMAIL }}
          node-version: 18 #optional, node-version you want to use following the Semantic Versioning Specification, default to current
          config: "firebase.json" #optional
          function: true #optional & boolean
          hosting: true #optional & boolean
          project: my-project
```

2. When this runs in GitHub Actions, I get the output:

```
/opt/hostedtoolcache/node/18.19.0/x64/bin/firebase deploy -m 18ee03a741bcc63fcbb7f22b8587449747537554 --config firebase.json --project my-project --only function hosting
Error: An error occured while deploying to Firebase: Error: The process '/opt/hostedtoolcache/node/18.19.0/x64/bin/firebase' failed with exit code 1. Retrying with debug mode enabled ...
/opt/hostedtoolcache/node/18.19.0/x64/bin/firebase deploy -m 18ee03a741bcc63fcbb7f22b8[58](https://github.com/<repo>/actions/runs/7978351224/job/21783788257#step:3:64)7449747537554 --config firebase.json --project my-project --only function hosting --debug
Error: An error occured while deploying to Firebase: Error: The process '/opt/hostedtoolcache/node/18.19.0/x64/bin/firebase' failed with exit code 1
Error: Process completed with exit code 1.
```

However, the logs from running the actual `firebase deploy` command are lost.

The entire log:
```
2024-02-20T18:51:02.5559176Z Requested labels: ubuntu-latest
2024-02-20T18:51:02.5559526Z Job defined at: <repo>/.github/workflows/firebase-merge.yaml@refs/heads/main
2024-02-20T18:51:02.5559640Z Waiting for a runner to pick up this job...
2024-02-20T18:51:03.0160910Z Job is waiting for a hosted runner to come online.
2024-02-20T18:51:06.1612600Z Job is about to start running on the hosted runner: GitHub Actions 1 (hosted)
2024-02-20T18:51:08.7764467Z Current runner version: '2.313.0'
2024-02-20T18:51:08.7804675Z ##[group]Operating System
2024-02-20T18:51:08.7806020Z Ubuntu
2024-02-20T18:51:08.7806516Z 22.04.3
2024-02-20T18:51:08.7807057Z LTS
2024-02-20T18:51:08.7807553Z ##[endgroup]
2024-02-20T18:51:08.7808103Z ##[group]Runner Image
2024-02-20T18:51:08.7808817Z Image: ubuntu-22.04
2024-02-20T18:51:08.7809386Z Version: 20240212.1.0
2024-02-20T18:51:08.7810800Z Included Software: https://github.com/actions/runner-images/blob/ubuntu22/20240212.1/images/ubuntu/Ubuntu2204-Readme.md
2024-02-20T18:51:08.7812931Z Image Release: https://github.com/actions/runner-images/releases/tag/ubuntu22%2F20240212.1
2024-02-20T18:51:08.7814112Z ##[endgroup]
2024-02-20T18:51:08.7814705Z ##[group]Runner Image Provisioner
2024-02-20T18:51:08.7815513Z 2.0.341.1
2024-02-20T18:51:08.7816044Z ##[endgroup]
2024-02-20T18:51:08.7817355Z ##[group]GITHUB_TOKEN Permissions
2024-02-20T18:51:08.7819633Z Contents: read
2024-02-20T18:51:08.7820256Z Metadata: read
2024-02-20T18:51:08.7821214Z ##[endgroup]
2024-02-20T18:51:08.7825776Z Secret source: Actions
2024-02-20T18:51:08.7826630Z Prepare workflow directory
2024-02-20T18:51:08.8918626Z Prepare all required actions
2024-02-20T18:51:08.9222665Z Getting action download info
2024-02-20T18:51:09.0936476Z Download action repository 'actions/checkout@v3' (SHA:f43a0e5ff2bd294095638e18286ca9a3d1956744)
2024-02-20T18:51:09.4292314Z Download action repository 'sws2apps/firebase-deployment@main' (SHA:dedec6e61fe462b8e71902b5b7f0c80f34a3a99c)
2024-02-20T18:51:09.9349780Z Getting action download info
2024-02-20T18:51:10.0491703Z Download action repository 'actions/setup-node@60edb5dd545a775178f52524783378180af0d1f8' (SHA:60edb5dd545a775178f52524783378180af0d1f8)
2024-02-20T18:51:10.2534129Z Download action repository 'google-github-actions/auth@a6e2e39c0a0331da29f7fd2c2a20a427e8d3ad1f' (SHA:a6e2e39c0a0331da29f7fd2c2a20a427e8d3ad1f)
2024-02-20T18:51:10.5451247Z Complete job name: Build and Deploy
2024-02-20T18:51:10.6759569Z ##[group]Run actions/checkout@v3
2024-02-20T18:51:10.6760617Z with:
2024-02-20T18:51:10.6761031Z   repository: <repo>
2024-02-20T18:51:10.6762051Z   token: ***
2024-02-20T18:51:10.6762440Z   ssh-strict: true
2024-02-20T18:51:10.6762854Z   persist-credentials: true
2024-02-20T18:51:10.6763313Z   clean: true
2024-02-20T18:51:10.6763712Z   sparse-checkout-cone-mode: true
2024-02-20T18:51:10.6764411Z   fetch-depth: 1
2024-02-20T18:51:10.6764951Z   fetch-tags: false
2024-02-20T18:51:10.6765330Z   lfs: false
2024-02-20T18:51:10.6765695Z   submodules: false
2024-02-20T18:51:10.6766115Z   set-safe-directory: true
2024-02-20T18:51:10.6766568Z ##[endgroup]
2024-02-20T18:51:11.1046127Z Syncing repository: <repo>
2024-02-20T18:51:11.1049788Z ##[group]Getting Git version info
2024-02-20T18:51:11.1051809Z Working directory is '/home/runner/work/<directory>'
2024-02-20T18:51:11.1053510Z [command]/usr/bin/git version
2024-02-20T18:51:11.1176438Z git version 2.43.0
2024-02-20T18:51:11.1229175Z ##[endgroup]
2024-02-20T18:51:11.1271082Z Temporarily overriding HOME='/home/runner/work/_temp/56f7e8cb-f11e-4e1c-9cde-9ddf0d686867' before making global git config changes
2024-02-20T18:51:11.1273366Z Adding repository directory to the temporary git global config as a safe directory
2024-02-20T18:51:11.1275082Z [command]/usr/bin/git config --global --add safe.directory /home/runner/work/<directory>
2024-02-20T18:51:11.1342884Z Deleting the contents of '/home/runner/work/<directory>'
2024-02-20T18:51:11.1357285Z ##[group]Initializing the repository
2024-02-20T18:51:11.1358315Z [command]/usr/bin/git init /home/runner/work/<directory>
2024-02-20T18:51:11.1527103Z hint: Using 'master' as the name for the initial branch. This default branch name
2024-02-20T18:51:11.1529006Z hint: is subject to change. To configure the initial branch name to use in all
2024-02-20T18:51:11.1530597Z hint: of your new repositories, which will suppress this warning, call:
2024-02-20T18:51:11.1532158Z hint: 
2024-02-20T18:51:11.1533588Z hint: 	git config --global init.defaultBranch <name>
2024-02-20T18:51:11.1534518Z hint: 
2024-02-20T18:51:11.1535430Z hint: Names commonly chosen instead of 'master' are 'main', 'trunk' and
2024-02-20T18:51:11.1536883Z hint: 'development'. The just-created branch can be renamed via this command:
2024-02-20T18:51:11.1537766Z hint: 
2024-02-20T18:51:11.1538192Z hint: 	git branch -m <name>
2024-02-20T18:51:11.1560552Z Initialized empty Git repository in /home/runner/work/<directory>/.git/
2024-02-20T18:51:11.1576198Z [command]/usr/bin/git remote add origin https://github.com/<repo>
2024-02-20T18:51:11.1636485Z ##[endgroup]
2024-02-20T18:51:11.1653243Z ##[group]Disabling automatic garbage collection
2024-02-20T18:51:11.1655084Z [command]/usr/bin/git config --local gc.auto 0
2024-02-20T18:51:11.1695617Z ##[endgroup]
2024-02-20T18:51:11.1697704Z ##[group]Setting up auth
2024-02-20T18:51:11.1711878Z [command]/usr/bin/git config --local --name-only --get-regexp core\.sshCommand
2024-02-20T18:51:11.1780828Z [command]/usr/bin/git submodule foreach --recursive sh -c "git config --local --name-only --get-regexp 'core\.sshCommand' && git config --local --unset-all 'core.sshCommand' || :"
2024-02-20T18:51:11.2336190Z [command]/usr/bin/git config --local --name-only --get-regexp http\.https\:\/\/github\.com\/\.extraheader
2024-02-20T18:51:11.2373493Z [command]/usr/bin/git submodule foreach --recursive sh -c "git config --local --name-only --get-regexp 'http\.https\:\/\/github\.com\/\.extraheader' && git config --local --unset-all 'http.https://github.com/.extraheader' || :"
2024-02-20T18:51:11.2703341Z [command]/usr/bin/git config --local http.https://github.com/.extraheader AUTHORIZATION: basic ***
2024-02-20T18:51:11.2790159Z ##[endgroup]
2024-02-20T18:51:11.2793575Z ##[group]Fetching the repository
2024-02-20T18:51:11.2808793Z [command]/usr/bin/git -c protocol.version=2 fetch --no-tags --prune --progress --no-recurse-submodules --depth=1 origin +18ee03a741bcc63fcbb7f22b8587449747537554:refs/remotes/origin/main
2024-02-20T18:51:11.5683611Z remote: Enumerating objects: 708, done.        
2024-02-20T18:51:11.5686345Z remote: Counting objects:   0% (1/708)
...
2024-02-20T18:51:11.5800444Z remote: Counting objects: 100% (708/708), done.        
2024-02-20T18:51:11.5801710Z remote: Compressing objects:   0% (1/574)
...
2024-02-20T18:51:12.3941196Z remote: Compressing objects: 100% (574/574), done.        
2024-02-20T18:51:12.3980620Z Receiving objects:   0% (1/708)
...
2024-02-20T18:51:15.4040516Z Receiving objects: 100% (708/708), 127.44 MiB | 42.38 MiB/s, done.
2024-02-20T18:51:15.4052516Z Resolving deltas:   0% (0/158)
...
2024-02-20T18:51:15.4458854Z Resolving deltas: 100% (158/158), done.
2024-02-20T18:51:18.7653748Z From <repo>
2024-02-20T18:51:18.7655359Z  * [new ref]         18ee03a741bcc63fcbb7f22b8587449747537554 -> origin/main
2024-02-20T18:51:18.7694102Z ##[endgroup]
2024-02-20T18:51:18.7694890Z ##[group]Determining the checkout info
2024-02-20T18:51:18.7695844Z ##[endgroup]
2024-02-20T18:51:18.7696728Z ##[group]Checking out the ref
2024-02-20T18:51:18.7701818Z [command]/usr/bin/git checkout --progress --force -B main refs/remotes/origin/main
2024-02-20T18:51:19.6083366Z Switched to a new branch 'main'
2024-02-20T18:51:19.6088633Z branch 'main' set up to track 'origin/main'.
2024-02-20T18:51:19.6125365Z ##[endgroup]
2024-02-20T18:51:19.6197214Z [command]/usr/bin/git log -1 --format='%H'
2024-02-20T18:51:19.6245999Z '18ee03a741bcc63fcbb7f22b8587449747537554'
2024-02-20T18:51:19.6793154Z ##[group]Run sws2apps/firebase-deployment@main
2024-02-20T18:51:19.6793877Z with:
2024-02-20T18:51:19.6795464Z   workloadIdentityProvider: ***
2024-02-20T18:51:19.6796261Z   serviceAccount: ***
2024-02-20T18:51:19.6796620Z   node-version: 18
2024-02-20T18:51:19.6796965Z   config: firebase.json
2024-02-20T18:51:19.6797328Z   function: true
2024-02-20T18:51:19.6797646Z   hosting: true
2024-02-20T18:51:19.6798189Z   project: my-project
2024-02-20T18:51:19.6798760Z ##[endgroup]
2024-02-20T18:51:19.7183288Z ##[group]Run actions/setup-node@60edb5dd545a775178f52524783378180af0d1f8
2024-02-20T18:51:19.7184328Z with:
2024-02-20T18:51:19.7184911Z   node-version: 18
2024-02-20T18:51:19.7185606Z   always-auth: false
2024-02-20T18:51:19.7186263Z   check-latest: false
2024-02-20T18:51:19.7187192Z   token: ***
2024-02-20T18:51:19.7187696Z ##[endgroup]
2024-02-20T18:51:20.3006967Z Found in cache @ /opt/hostedtoolcache/node/18.19.0/x64
2024-02-20T18:51:20.3032484Z ##[group]Environment details
2024-02-20T18:51:21.2457753Z node: v18.19.0
2024-02-20T18:51:21.2458346Z npm: 10.2.3
2024-02-20T18:51:21.2458753Z yarn: 1.22.21
2024-02-20T18:51:21.2461829Z ##[endgroup]
2024-02-20T18:51:21.2894939Z ##[group]Run npm install -g firebase-tools
2024-02-20T18:51:21.2895573Z [36;1mnpm install -g firebase-tools[0m
2024-02-20T18:51:21.2946582Z shell: /usr/bin/bash --noprofile --norc -e -o pipefail {0}
2024-02-20T18:51:21.2947169Z ##[endgroup]
2024-02-20T18:51:43.6827351Z 
2024-02-20T18:51:43.6837229Z added 640 packages in 22s
2024-02-20T18:51:43.6837907Z 
2024-02-20T18:51:43.6838292Z 66 packages are looking for funding
2024-02-20T18:51:43.6838920Z   run `npm fund` for details
2024-02-20T18:51:43.7220432Z ##[group]Run google-github-actions/auth@a6e2e39c0a0331da29f7fd2c2a20a427e8d3ad1f
2024-02-20T18:51:43.7221153Z with:
2024-02-20T18:51:43.7222092Z   workload_identity_provider: ***
2024-02-20T18:51:43.7223131Z   service_account: ***
2024-02-20T18:51:43.7223500Z   create_credentials_file: true
2024-02-20T18:51:43.7224102Z   export_environment_variables: true
2024-02-20T18:51:43.7224559Z   universe: googleapis.com
2024-02-20T18:51:43.7225316Z   cleanup_credentials: true
2024-02-20T18:51:43.7225917Z   access_token_lifetime: 3600s
2024-02-20T18:51:43.7226592Z   access_token_scopes: https://www.googleapis.com/auth/cloud-platform
2024-02-20T18:51:43.7227481Z   id_token_include_email: false
2024-02-20T18:51:43.7228415Z ##[endgroup]
2024-02-20T18:51:43.9953389Z Created credentials file at "/home/runner/work/<directory>/gha-creds-d05d4f4af96dadf1.json"
2024-02-20T18:51:44.1092432Z ##[group]Run node /home/runner/work/_actions/sws2apps/firebase-deployment/main/dist/index.js
2024-02-20T18:51:44.1093548Z [36;1mnode /home/runner/work/_actions/sws2apps/firebase-deployment/main/dist/index.js[0m
2024-02-20T18:51:44.1119867Z shell: /usr/bin/bash --noprofile --norc -e -o pipefail {0}
2024-02-20T18:51:44.1120477Z env:
2024-02-20T18:51:44.1121846Z   CLOUDSDK_AUTH_CREDENTIAL_FILE_OVERRIDE: /home/runner/work/<directory>/gha-creds-d05d4f4af96dadf1.json
2024-02-20T18:51:44.1123277Z   GOOGLE_APPLICATION_CREDENTIALS: /home/runner/work/<directory>/gha-creds-d05d4f4af96dadf1.json
2024-02-20T18:51:44.1124596Z   GOOGLE_GHA_CREDS_PATH: /home/runner/work/<repo>/gha-creds-d05d4f4af96dadf1.json
2024-02-20T18:51:44.1126799Z   CLOUDSDK_CORE_PROJECT: my-project
2024-02-20T18:51:44.1127464Z   CLOUDSDK_PROJECT: my-project
2024-02-20T18:51:44.1128518Z   GCLOUD_PROJECT: my-project
2024-02-20T18:51:44.1129457Z   GCP_PROJECT: my-project
2024-02-20T18:51:44.1130322Z   GOOGLE_CLOUD_PROJECT: my-project
2024-02-20T18:51:44.1131674Z   project: my-project
2024-02-20T18:51:44.1132132Z   config: firebase.json
2024-02-20T18:51:44.1132555Z   function: true
2024-02-20T18:51:44.1132938Z   hosting: true
2024-02-20T18:51:44.1133310Z ##[endgroup]
2024-02-20T18:51:44.2096781Z [command]/opt/hostedtoolcache/node/18.19.0/x64/bin/firebase deploy -m 18ee03a741bcc63fcbb7f22b8587449747537554 --config firebase.json --project my-project --only function hosting
2024-02-20T18:51:46.5873519Z ##[error]An error occured while deploying to Firebase: Error: The process '/opt/hostedtoolcache/node/18.19.0/x64/bin/firebase' failed with exit code 1. Retrying with debug mode enabled ...
2024-02-20T18:51:46.5877764Z [command]/opt/hostedtoolcache/node/18.19.0/x64/bin/firebase deploy -m 18ee03a741bcc63fcbb7f22b8587449747537554 --config firebase.json --project my-project --only function hosting --debug
2024-02-20T18:51:48.8096831Z ##[error]An error occured while deploying to Firebase: Error: The process '/opt/hostedtoolcache/node/18.19.0/x64/bin/firebase' failed with exit code 1
2024-02-20T18:51:48.8156714Z ##[error]Process completed with exit code 1.
2024-02-20T18:51:48.8327757Z Post job cleanup.
2024-02-20T18:51:48.8409888Z Post job cleanup.
2024-02-20T18:51:48.9700654Z Removed exported credentials at "/home/runner/work/<directory>/gha-creds-d05d4f4af96dadf1.json".
2024-02-20T18:51:49.0037629Z Post job cleanup.
2024-02-20T18:51:49.1424054Z [command]/usr/bin/git version
2024-02-20T18:51:49.1487186Z git version 2.43.0
2024-02-20T18:51:49.1559823Z Temporarily overriding HOME='/home/runner/work/_temp/be3f9981-bdf8-45a0-8bbe-bdd7cfcb1742' before making global git config changes
2024-02-20T18:51:49.1565005Z Adding repository directory to the temporary git global config as a safe directory
2024-02-20T18:51:49.1575226Z [command]/usr/bin/git config --global --add safe.directory /home/runner/work/<directory>
2024-02-20T18:51:49.1668005Z [command]/usr/bin/git config --local --name-only --get-regexp core\.sshCommand
2024-02-20T18:51:49.1738737Z [command]/usr/bin/git submodule foreach --recursive sh -c "git config --local --name-only --get-regexp 'core\.sshCommand' && git config --local --unset-all 'core.sshCommand' || :"
2024-02-20T18:51:49.2338734Z [command]/usr/bin/git config --local --name-only --get-regexp http\.https\:\/\/github\.com\/\.extraheader
2024-02-20T18:51:49.2340203Z http.https://github.com/.extraheader
2024-02-20T18:51:49.2344020Z [command]/usr/bin/git config --local --unset-all http.https://github.com/.extraheader
2024-02-20T18:51:49.2359760Z [command]/usr/bin/git submodule foreach --recursive sh -c "git config --local --name-only --get-regexp 'http\.https\:\/\/github\.com\/\.extraheader' && git config --local --unset-all 'http.https://github.com/.extraheader' || :"
2024-02-20T18:51:49.2983118Z Cleaning up orphan processes
```

**Expected behavior**
The GitHub Actions toolkit actually offers a [getExecOutput function](https://github.com/actions/toolkit/blob/main/packages/exec/src/exec.ts#L44), which returns a promise containing the logs from the underlying exec'd command.  It would be great if your code could call that instead, and print the contents of the log to stdout and stderr.

**Screenshots**
N/A

**Desktop (please complete the following information):**
N/A

**Smartphone (please complete the following information):**
N/A

**Additional context**
N/A

--- a/src/index.js
+++ b/src/index.js
@@ -3,6 +3,8 @@ import core from '@actions/core';
 import exec from '@actions/exec';
 
 const run = async () => {
+	core.info("This is the joinsunfish version of the code.");
+
 	// preflight check before starting the actions
 	const project = process.env.project;
 	if (!project) {

--- a/src/index.js
+++ b/src/index.js
@@ -24,26 +24,43 @@ const run = async () => {
 		deployList.push('hosting');
 	}
 
-	const deployOnly = deployList.join(',');
+	let args = ['deploy', '-m', process.env.GITHUB_SHA];
 
-	let cmd = `firebase deploy -m ${process.env.GITHUB_SHA}`;
-	cmd += config ? ` --config ${config}` : ' ';
-	cmd += ` --project ${project}`;
-	cmd += deployOnly !== '' ? ` --only ${deployOnly}` : '';
+	if (config) {
+		args.push('--config', config);
+	}
+
+	args.push('--project', project);
+
+	if (deployOnly.length > 0) {
+		args.push('--only', deployList.join(','));
+	}
 
 	try {
+		const options = {};
+		options.listeners = {
+		  stdout: (data) => {
+			process.stdout.write(data.toString());
+		  },
+		  stderr: (data) => {
+			process.stderr.write(data.toString());
+		  },
+		  stdline: (data) => {
+			process.stdline.write(data.toString());
+		  },
+		};
+
 		// attempt to run firebase deploy, and run with debug mode if failed
-		// TODO:
-		//   1. Add the current stdout and stderr as listeners to the getExecOutput listeners https://github.com/actions/toolkit/blob/main/packages/exec/src/exec.ts#L44
-		//   2. Print the result output using the relevant core function.
-		const result = await exec.getExecOutput(cmd);
+		const result = await exec.exec("firebase", args, options);
 
 	} catch (error) {
 		core.error(`An error occured while deploying to Firebase: ${error}. Retrying with debug mode enabled ...`);
 
 		// attempt to run firebase deploy with debug mode
+		args.push("--debug");
+
 		try {
-			await exec.exec(`${cmd} --debug`);
+			await exec.exec("firebase", args, options);
 		} catch (error) {
 			core.setFailed(`An error occured while deploying to Firebase: ${error}`);
 		}

--- a/src/index.js
+++ b/src/index.js
@@ -39,20 +39,20 @@ const run = async () => {
 		args.push('--only', deployList.join(','));
 	}
 
-	try {
-		const options = {};
-		options.listeners = {
-		  stdout: (data) => {
-			process.stdout.write(data.toString());
-		  },
-		  stderr: (data) => {
-			process.stderr.write(data.toString());
-		  },
-		  stdline: (data) => {
-			process.stdline.write(data.toString());
-		  },
-		};
+	const options = {};
+	options.listeners = {
+	  stdout: (data) => {
+		process.stdout.write(data.toString());
+	  },
+	  stderr: (data) => {
+		process.stderr.write(data.toString());
+	  },
+	  stdline: (data) => {
+		process.stdline.write(data.toString());
+	  },
+	};
 
+	try {
 		// attempt to run firebase deploy, and run with debug mode if failed
 		const result = await exec.exec("firebase", args, options);
 

--- a/src/index.js
+++ b/src/index.js
@@ -14,9 +14,17 @@ const run = async () => {
 	const config = process.env.config;
 
 	// check only deployment settings
-	let deployOnly = process.env.function === 'true' ? 'function' : '';
-	deployOnly += deployOnly === '' ? '' : ' ';
-	deployOnly += process.env.hosting === 'true' ? `hosting` : '';
+	let deployList = [];
+
+	if (process.env.function === 'true') {
+		deployList.push('functions');
+	}
+
+	if (process.env.hosting === 'true') {
+		deployList.push('hosting');
+	}
+
+	const deployOnly = deployList.join(',');
 
 	let cmd = `firebase deploy -m ${process.env.GITHUB_SHA}`;
 	cmd += config ? ` --config ${config}` : ' ';
@@ -25,7 +33,11 @@ const run = async () => {
 
 	try {
 		// attempt to run firebase deploy, and run with debug mode if failed
-		await exec.exec(cmd);
+		// TODO:
+		//   1. Add the current stdout and stderr as listeners to the getExecOutput listeners https://github.com/actions/toolkit/blob/main/packages/exec/src/exec.ts#L44
+		//   2. Print the result output using the relevant core function.
+		const result = await exec.getExecOutput(cmd);
+
 	} catch (error) {
 		core.error(`An error occured while deploying to Firebase: ${error}. Retrying with debug mode enabled ...`);
 

--- a/src/index.js
+++ b/src/index.js
@@ -37,7 +37,7 @@ const run = async () => {
 
 	args.push('--project', project);
 
-	if (deployOnly.length > 0) {
+	if (deployList.length > 0) {
 		args.push('--only', deployList.join(','));
 	}
 

--- a/src/index.js
+++ b/src/index.js
@@ -3,11 +3,6 @@ import core from '@actions/core';
 import exec from '@actions/exec';
 
 const run = async () => {
-	core.info("This is the joinsunfish version of the code.");
-	core.error("IS ANYONE LISTENING TO ME?!");
-
-	process.stdout.write("HELLO?!\n");
-
 	// preflight check before starting the actions
 	const project = process.env.project;
 	if (!project) {
@@ -29,13 +24,11 @@ const run = async () => {
 		deployList.push('hosting');
 	}
 
-	let args = ['deploy', '-m', process.env.GITHUB_SHA];
+	let args = ['deploy', '-m', process.env.GITHUB_SHA, '--project', project];
 
 	if (config) {
 		args.push('--config', config);
 	}
-
-	args.push('--project', project);
 
 	if (deployList.length > 0) {
 		args.push('--only', deployList.join(','));

--- a/src/index.js
+++ b/src/index.js
@@ -6,6 +6,8 @@ const run = async () => {
 	core.info("This is the joinsunfish version of the code.");
 	core.error("IS ANYONE LISTENING TO ME?!");
 
+	process.stdout.write("HELLO?!\n");
+
 	// preflight check before starting the actions
 	const project = process.env.project;
 	if (!project) {

--- a/src/index.js
+++ b/src/index.js
@@ -49,7 +49,7 @@ const run = async () => {
 
 	try {
 		// attempt to run firebase deploy, and run with debug mode if failed
-		const result = await exec.exec("firebase", args, options);
+		await exec.exec("firebase", args, options);
 
 	} catch (error) {
 		core.error(`An error occured while deploying to Firebase: ${error}. Retrying with debug mode enabled ...`);

--- a/src/index.js
+++ b/src/index.js
@@ -4,6 +4,7 @@ import exec from '@actions/exec';
 
 const run = async () => {
 	core.info("This is the joinsunfish version of the code.");
+	core.error("IS ANYONE LISTENING TO ME?!");
 
 	// preflight check before starting the actions
 	const project = process.env.project;

